Makefile

```diff
@@ -0,0 +1,29 @@
+# Makefile
+THIS_FILE := $(lastword $(MAKEFILE_LIST))
+include common.mk
+SHELL := /bin/bash
+
+RUNTEST=python -m unittest -v -b
+
+ALLMODULES=$(patsubst %.py, %, $(wildcard test_*.py))
+
+all:
+	@echo "running tests..."
+	${RUNTEST} ${ALLMODULES}
+
+% : test_%.py
+	${RUNTEST} test_$@
+
+venv: venv/bin/activate
+
+venv/bin/activate: requirements.txt
+	test -d .venvbag || virtualenv --python=python3.6 .venvbag
+	.venvbag/bin/pip install -Ur requirements.txt
+	touch .venvbag/bin/activate
+
+deploy: venv
+	@$(MAKE) -f $(THIS_FILE)  all
+	.venvbag/bin/pip install awscli --upgrade
+	@echo "deploying the Lambda"
+	.venvbag/bin/chalice deploy --stage dev
+	.venvbag/bin/chalice deploy --stage staging
```

Comment:
I think you should have separate deploys for dev and staging. The typical flow would be to deploy to dev repeatedly until you are sure it is working correctly, then deploy to staging when you are confident it works.

Reply:
do you mean like a retry on the `chalice deploy`? perhaps it does it already internally? I can research it. I can separate deployment of the two stages.

Reply:
I mean you should have separate targets to deploy dev and to deploy stage.